### PR TITLE
[FLOC-3882] Docker plugin: add support for VolumeDriver.List and VolumeDriver.Get

### PR DESCRIPTION
**Pull request by itamarst**
`docker volume ls` and `docker volume inspect` now use the Docker volume plugin API.

This adds support for the two new calls:
https://github.com/docker/docker/blob/master/docs/extend/plugins_volume.md#volumedriverget

Note that at some point Docker changed their docs to say one should return `"Err": ""` for non-errors; it used to say `"Err": null`. I will do follow up branch to address this for existing commands, but did the new way for the new commands.

I ran the new acceptance tests with an experimental Docker build (`1.10.0-dev`):

```
$ trial flocker.acceptance.endtoend.test_dockerplugin.DockerPluginTests.test_{listed,inspect}
flocker.acceptance.endtoend.test_dockerplugin
  DockerPluginTests
    test_listed ...                                                        [OK]
    test_inspect ...                                                       [OK]

-------------------------------------------------------------------------------
Ran 2 tests in 84.871s

PASSED (successes=2)
```

### CODE REVIEW

**Comment by itamarst** [comment]
I'm seeing test failures for other acceptance tests when running against the experimental build, so will attempt to debug and see what's going. Possibly a different issue, possibly just that it's experimental, possibly this branch has a bug.

**Comment by itamarst** [comment]
Looks like Docker access control is hosed for this build. Not clear how or why, presumably something to do with AppArmor:

```
root@ip-172-31-11-187:~# docker run busybox ls /tmp
ls: /tmp: Permission denied
```

**Comment by itamarst** [comment]
Removing apparmor profile seemed to help; re-running acceptance tests.

**Comment by itamarst** [comment]
Mostly they passed; couple more failures are due to having old version of `docker-py` so rerunning.

**Comment by itamarst** [comment]
Yep, all good now. I am skeptical AppArmor problem has anything to do with this branch. So: with 1.10-dev all acceptance tests pass with this branch.

File: flocker/acceptance/endtoend/test_dockerplugin.py [hunk 1]
```diff
@@ -446,3 +446,47 @@ def test_move_volume_different_node(self, cluster):
...
+        """
+        A volume created outside of Docker can be inspected via the Docker
+        Volume API.
+
+        :param cluster: Flocker cluster.
+        """
+        self.require_docker('1.10.0', cluster)
+        name = random_name(self)
+
+        d = create_dataset(self, cluster, metadata={u"name": name})
```
Comment (jml) [inline flocker/acceptance/endtoend/test_dockerplugin.py]: :question: Maybe use `NAME_FIELD` in our tests, too?

**Comment by jml** [comment]
The AppArmor problems needn't block this patch, agreed. Are they likely to come up again? I'm keen that no-one else need solve the same problem twice.

Please feel free to merge this after addressing comments & ensuring tests pass.

**Comment by itamarst** [comment]
No idea if AppArmor will come up again. Hopefully either it was specific to something I did, or the Docker packages are broken and hopefully someone will report bug against 1.10rc1.

**Pull request merged**